### ISSUE TITLE
AP-2rep motif

### ISSUE THREAD

**Issue by iamjli** [fraenkel-lab/GarNet#21]
This motif is being included despite having scores of 0. 

![image](https://cloud.githubusercontent.com/assets/7219365/24280494/5dced66e-1027-11e7-8e98-c64e8429e218.png)

**Comment by zfrenchee** [comment]
Is this causing a bug? Seems like it. If so, that means we need to add a line of code which removes all motifs with a score of 0 in the `parse_motifs_file` function. I'm happy to do that tomorrow, or help you do it Johnny. 

@iamjli @AmandaKedaigle I think it's fair to say this new GarNet is a brittle because it hasn't been battle-hardened with use. As we go, we'll find fewer and fewer bugs. I still believe it represents at least a modest improvement, so I hope you're willing to go through this with me, ironing out each bug as it comes up. 

One way of accelerating this process is by writing tests. I'm learning travis right now via my work on Ludwig's pcst_fast, so maybe I'll add tests here to try to locate a batch of these potential issues all at once. Let me know what you think.

**Comment by agitter** [comment]
@zfrenchee I'm not especially knowledgeable, but I have some experience with Travis CI.  If you have specific questions, let me know.

**Comment by zfrenchee** [comment]
@agitter here's the question: http://stackoverflow.com/questions/42986906/supporting-two-languages-in-travis-ci-file

**Comment by AmandaKedaigle** [comment]
@zfrenchee Absolutely! I would not expect anything less than finding a bunch of bugs as we start to use it, which is why I was advocating doing this process before calling anything done and letting it out of our little bubble. Tests is a great idea. While it's going to be an improvement, it's not yet, so let's get it there first :)

**Comment by agitter** [comment]
That's a question I've had myself on a different project but not ever had properly answered.  My first instinct was to set up two different continuous integration servers.  That seemed like a bad idea so I didn't follow through.

Do you want to be able to run tests in both Python and C++ or is the objective to run Python tests that require the C++ dependency?  If the latter, you can follow the [Omics Integrator example](https://github.com/fraenkel-lab/OmicsIntegrator/blob/bb0b881f8de02db683c046e4634eee4e228b63c9/.travis.yml#L27).  It builds msgsteiner and the proceeds to run the Python tests, which indirectly test that msgsteiner is functioning properly.  It's still a hack so I'm curious if you figure out a better solution.  Ideally the C++ code would have its own test suite in its own repository.

**Comment by zfrenchee** [comment]
@agitter I think our code has both important c++ and python test which are somewhat independent -- the c++ tests are more "unit test"-y and the python tests are "integration test"-y. I'd like to have them both called from the same file. I'll send that question over to some gurus. ;)

@iamjli would you be willing to submit a PR to fix this bug you encountered, once you get the chance?
